ephios/static/ephios/js/serviceworker.js

```diff
@@ -19,40 +8,86 @@ const filesToCache = [
     "/static/ephios/img/ephios-text-black.png",
 ];
 
-// Cache on install
 self.addEventListener("install", event => {
-    this.skipWaiting();
     event.waitUntil(
-        caches.open(staticCacheName)
-            .then(cache => {
-                return cache.addAll(filesToCache);
-            })
-    )
+        caches.open(staticCacheName).then(cache => {
+            return cache.addAll(staticFilesToCacheOnInstall).then(() => {
+                this.skipWaiting();
+            });
+        })
+    );
 });
 
-// Clear cache on activate
 self.addEventListener('activate', event => {
     event.waitUntil(
         caches.keys().then(cacheNames => {
             return Promise.all(
                 cacheNames
-                    .filter(cacheName => (cacheName.startsWith("django-pwa-")))
+                    .filter(cacheName => (cacheName.startsWith("ephios-pwa-")))
                     .filter(cacheName => (cacheName !== staticCacheName))
                     .map(cacheName => caches.delete(cacheName))
             );
         })
     );
 });
 
-// Serve from Cache
+async function markResponseAsOffline(response) {
+    let content = await response.body.getReader().read().then(r => r.value);
```

Comment:
I think this line is a great demonstration of javascripts async API.
`await` obvs awaits a Promise to be fulfilled, but look how this is actually a chained promise because of the `then` at the end.